tests/test_downloadermiddleware_httpcompression.py

```diff
@@ -173,29 +174,71 @@ def test_process_response_gzipped_contenttype(self):
         request = response.request
 
         newresponse = self.mw.process_response(request, response, self.spider)
-        self.assertIs(newresponse, response)
-        self.assertEqual(response.headers['Content-Encoding'], b'gzip')
-        self.assertEqual(response.headers['Content-Type'], b'application/gzip')
+        assert newresponse is not response
+        assert newresponse.body.startswith(b'<!DOCTYPE')
+        assert 'Content-Encoding' not in newresponse.headers
```

Comment:
pytest magic asserts are disabled in scrapy testing suite (they break contracts tests if I recall it correctly), so it is better to keep assertIs / assertEqual